src/common/box_filters.c

```diff
@@ -493,7 +493,7 @@ static void dt_box_mean_1ch(float *const buf, const int height, const int width,
                             const int iterations)
 {
   const int size = MAX(width,height);
-  float *const restrict scanlines = dt_alloc_align(64, 4 * size * sizeof(float) * dt_get_num_threads());
+  float *const restrict scanlines = dt_alloc_align_float((size_t)4 * size * dt_get_num_threads());
```

Comment:
alternatively, you could redefine dt_get_num_threads as
`static inline size_t dt_get_num_threads()`
(and reorder the multiplication)

That would get rid of a fair number of the explicit `(size_t)` casts. And is arguably a more logical interface.

Reply:
while I do agree, the `dt_get_num_threads` is basically a call to `omp_get_num_procs` which returns int :/ So I'm down to changing def to size_t if I get a buy-in... @TurboGit or @ralfbrown or others :)

Reply:
I'm fine with casting the result from `omp_get_num_procs`.  If it ever returns a value < 1, lots of stuff in dt will blow up.  And it will be a *very* long time until there's a shared-memory machine with more than 2 billion CPUs....